### ISSUE TITLE
[Question] How to test the existence of a locator in an iframe

### ISSUE THREAD

**Issue by Nosetiduno** [microsoft/playwright#19454]
I try to test if a locator is in an iframe with these lines of code : 

```
let boolExist = false;

element = await page.frameLocator(frame).locator(locator);

if (await element.count() > 0) {
    boolExist = true;
}

```

It works when the locator is in the iframe but when it is not in, I get a time out error.

These lines of code work when the locator is not in an iframe. When the locator is not in the dom the function count return 0 and there is not time out error.

Do you know how can I test if a locator is present in an iframe without getting a time out error if the locator is not present ?

Thank you for your answers

**Comment by yury-s** [comment]
`count()` returns without waiting for the elements, e.g. try this example:

```ts
import { test, expect } from '@playwright/test';

test('should return count without waiting', async ({ page }) => {
  await page.setContent('<iframe srcdoc="<p>Hello world!</p>"></iframe>');
  expect(await page.frameLocator("iframe").locator('p').count()).toBe(1);
  expect(await page.frameLocator("iframe").locator('div').count()).toBe(0);
});
```

Can you share a reproducible example where it fails for you?

**Comment by Nosetiduno** [comment]
The problem is that when I click on an locator , I don't know if  a pop up will appear or not and I try to detect if the pop up appear or not with an locator of it. But if the element is not present, I dont want the execution to stop (like the Exists function in UFT).

So as to detect if the element is present or not I have done this function : 

```
export async function testExistence (page, frame, locator) {
    let element = "";
    let boolExist = false;


    if (frame !== "") {
        element = await page.frameLocator(frame).locator(locator);
    } else {
        element = await page.locator(locator);
    }

       // it does not work when element is not found in an iframe. I get time out error
        if (await element.count() > 0) { 
            boolExist = true;
        }

    return boolExist;
}
```
The problem of this function is that when an element is not found in an iframe, I get a time out error instead count() function returning 0 like the case when an element is not in an iframe.

The main problem is that the application of my customer can be a little slow and I try to temporize the exection as I can :/

**Comment by yury-s** [comment]
> The main problem is that the application of my customer can be a little slow and I try to temporize the exection as I can :/
> I have tried to temporize with waitFor() functions on element I get time out error or put some lines like
await this.page.waitForLoadState('domcontentloaded');

This will likely result in a flaky behavior. You should figure out some condition which will tell you that by this time the popup should have been shown and you can check for it. The condition is usually application specific, it can be some javascript state or a UI element which appears in both scenarios and only after the file has been processed.

> The problem of this function is that when an element is not found in an iframe, I get a time out error instead count() function returning 0 like the case when an element is not in an iframe.

You cannot get timeout error from `await element.count()`, see my example above where count==0 is returned immeidately even though the `div` is never in the document. To help you further we need a reproducible example which we could run locally and see the problem. Please file a new issue with a full repro and link to this one when you get back to it!
